staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go

```diff
@@ -30,17 +32,27 @@ type OpenAPI struct {
 	Config *common.Config
 }
 
-// Install adds the SwaggerUI webservice to the given mux.
-func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) {
+// Install adds the SwaggerUI webservice to the given mux. This function returns
+// the built static OpenAPI spec and the registered OpenAPI services to allow
+// further OpenAPI spec aggregation.
+func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) (openAPIService, openAPIVersionedService *handler.OpenAPIService, spec *spec.Swagger) {
+	var err error
+	// Record the static OpenAPI spec to allow further OpenAPI spec aggregation
+	// with this static spec on the registered OpenAPI services
+	spec, err = builder.BuildOpenAPISpec(c.RegisteredWebServices(), oa.Config)
+	if err != nil {
+		klog.Fatalf("Failed to build open api spec for root: %v", err)
```

Comment:
maybe I asked before: why no error return type instead of `Fatalf`? this looks hacky.

Reply:
copied from existing code https://github.com/kubernetes/kubernetes/blob/8b98e802eddb9f478ff7d991a2f72f60c165388a/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go#L37